src/main/java/edu/umsystem/InsertionSort.java

```diff
@@ -0,0 +1,39 @@
+package edu.umsystem;
+
+public class InsertionSort implements SortingAlgorithm {
+    SortShow arrayDisplay;
+
+    InsertionSort(SortShow arrayDisplay) {
+        this.arrayDisplay = arrayDisplay;
+    }
+
+    @Override
+    public String getName() {
+        return "Insertion Sort";
+    }
+
+    @Override
+    public void sort() {
+        for (int unsortedIndex = 1; unsortedIndex < arrayDisplay.total_number_of_lines; ++unsortedIndex) {
+            int firstUnsortedElement = arrayDisplay.lines_lengths[unsortedIndex];
+            insertInOrder(firstUnsortedElement, arrayDisplay.lines_lengths, 0, unsortedIndex - 1);
+        }
+    }
+
+    public void insertInOrder(int element, int[] a, int begin, int end) {
+        // Inserts elements into the sorted array elements a[begin] through a[end]
+
+        int index = end;
+        while (index >= begin && element < a[index]) {
+            a[index + 1] = a[index]; // make room
+            index--;
+
+            arrayDisplay.paintComponent(arrayDisplay.getGraphics());
```

Comment:
re the drawing thing: I think drawing after every change like you're doing is the right approach (tbh I skipped it in shell sort because it took too long and flickered way too much for my liking :p) though once we have all the demos merged I have an idea for how we can do it (the one I mentioned in the Discord).

We also probably want to draw even more often than every change (ie: every comparison too) because otherwise selection sort is just going to blow everything out of the water (since its worst case performs exactly n swaps) but with the way I'm imagining it, as long as we can use the `checkLessThan`, and `swap` methods of `arrayDisplay` (plus something else for replacement like you have to do in merge sort) the sorting algorithms themselves won't have to worry about drawing at all.